mlir/include/mlir/Dialect/Rock/IR/RockOps.td

```diff
@@ -44,19 +45,28 @@ class TensorOrMemRefRankOf<list<Type> allowedTypes, list<int> ranks> :
              /*summary=*/"",
              "::mlir::ShapedType">;
 
+class I32ArrayLength<int n> : ConfinedAttr<I32ArrayAttr, [ArrayCount<n>]>;
 
-def Rock_Conv2DOp :
-    Rock_Op<"conv2d", [DeclareOpInterfaceMethods<RockGemmWrapperInterface>]>,
-    Arguments<(ins TensorOrMemRefRankOf<[F32, F16, BF16, I8], [5]>:$filter,
+class Rock_Conv2DOpBase<string mnemonic> :
```

Comment:
Note: only forward convolution support I8.

Note 2: could that additional argument list be a template argument (of type `dag` to `Conv2DOpBase`?